ipa-core/src/net/transport.rs

```diff
@@ -152,30 +187,88 @@ impl HttpTransport {
             r.await
         }
     }
+}
+
+impl MpcHttpTransport {
+    #[must_use]
+    pub fn new(
+        http_runtime: IpaRuntime,
+        identity: HelperIdentity,
+        server_config: ServerConfig,
+        network_config: NetworkConfig<Helper>,
+        clients: [MpcHelperClient; 3],
+        handler: Option<HandlerRef<HelperIdentity>>,
+    ) -> (Self, MpcHelperServer<Helper>) {
+        let transport = Self::new_internal(http_runtime, identity, clients, handler);
+        let server = MpcHelperServer::new_mpc(&transport, server_config, network_config);
+        (transport, server)
+    }
+
+    fn new_internal(
+        http_runtime: IpaRuntime,
+        identity: HelperIdentity,
+        clients: [MpcHelperClient; 3],
+        handler: Option<HandlerRef<HelperIdentity>>,
+    ) -> Self {
+        let mut id_clients = HashMap::new();
```

Comment:
nit: `HashMap::from_iter` is nicer for this use case. Also you may consider using arrays here - default hasher in Rust is pretty heavyweight. It should work for both shard index and helper identity as these are just numbers and they are not sparse